base/deprecated.jl

```diff
@@ -1499,4 +1499,12 @@ end
 # Calling promote_op is likely a bad idea, so deprecate its convenience wrapper promote_eltype_op
 @deprecate promote_eltype_op(op, As...) promote_op(op, map(eltype, As)...)
 
+# These conversions should not be defined, see #19896
+Base.convert{T<:Number}(::Type{T},x::Dates.Period) = error("conversion between Periods and Numbers is not defined")
+Base.convert{T<:Dates.Period}(::Type{T},x::Real) = error("conversion between Periods and Numbers is not defined")
+@deprecate convert{R<:Real}(::Type{R},x::Dates.DateTime) R(Dates.datetime2rata(x))
+@deprecate convert{R<:Real}(::Type{R},x::Dates.Date)     R(Dates.datetime2rata(x))
+@deprecate convert(::Type{Dates.DateTime}, x::Real) Dates.rata2datetime(x)
+@deprecate convert(::Type{Dates.Date}, x::Real)     Dates.Date(Dates.rata2datetime(x))
```

Comment:
GitHub seems to have messed up the threading order here: the comments were made in the order me, @quinnj, and then my comment about `convert(Date, x)` giving the wrong answer in response to @quinnj. In other words, the problem I highlighted is still a concern (it's why the tests fail for this PR).

What should the API actually look like? What functions should we export?